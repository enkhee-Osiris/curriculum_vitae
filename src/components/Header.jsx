import { useCallback } from 'react';
// import avatar from '../images/avatar.png';

// TODO: add avatar
function Header() {
  const toggleTheme = useCallback(() => {
    if (document.documentElement.classList.contains('dark')) {
      document.documentElement.classList.remove('dark');
      document.documentElement.classList.add('light');
    } else if (document.documentElement.classList.contains('light')) {
      document.documentElement.classList.remove('light');
      document.documentElement.classList.add('dark');
    }
  }, []);

  return (
    <header className="flex flex-row justify-between">
      <div className="space-y-6">
        <h1 className="text-4xl font-thin dark:font-light dark:text-gray-300">
          Enkh-Erdene Bolormaa
        </h1>
        <h2 className="text-3xl font-thin dark:font-light dark:text-gray-300">
          Front-End developer
        </h2>
      </div>
      <div className="flex flex-col space-y-4">
        <a
          target="_parent"
          href="link"
          className="transition-all ease-in-out duration-100 cursor-pointer p-2 focus:ring-3 hover:ring-3 ring-black dark:ring-white">
          <span className="sr-only">Download</span>

          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeLinejoin="round">
            <path d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4" />
            <polyline points="7 10 12 15 17 10" />
            <line x1="12" y1="15" x2="12" y2="3" />
          </svg>
        </a>

        <button
          onClick={toggleTheme}
          type="button"
          className="transition-all ease-in-out duration-100 cursor-pointer p-2 focus:ring-3 hover:ring-3 ring-black dark:ring-white">
          <span className="sr-only">Theme toggle</span>

          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6 hidden dark:block"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeLinejoin="round">
            <circle cx="12" cy="12" r="5" />
            <line x1="12" y1="1" x2="12" y2="3" />
            <line x1="12" y1="21" x2="12" y2="23" />
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64" />
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78" />
            <line x1="1" y1="12" x2="3" y2="12" />
            <line x1="21" y1="12" x2="23" y2="12" />
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36" />
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22" />
          </svg>

          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            className="h-6 w-6 block dark:hidden"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeLinejoin="round">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z" />
          </svg>
        </button>
      </div>
    </header>
  );
}

export default Header;
