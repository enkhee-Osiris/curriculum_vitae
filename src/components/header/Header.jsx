import React from 'react';
import logo from './image.png'

function Header() {
  return (
    <header>
      <img src={logo} alt="logo" width="200px"/>
      <h2>Enkh-Erdene Bolormaa</h2>
      <h3>Full-Stack Web Developer</h3>
    </header>
  );
}

export default Header;
