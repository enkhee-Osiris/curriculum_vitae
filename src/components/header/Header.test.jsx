import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('<Header />', () => {
  it('should be defined', () => {
    expect(Header).toBeDefined();
  });

  it('renders correctly', () => {
    const header = shallow(<Header />);

    expect(header).toMatchSnapshot();
  });

  it('renders img', () => {
    const header = shallow(<Header />);

    expect(header.find('img').length).toBeGreaterThan(0);
  });

  it('renders h2', () => {
    const header = shallow(<Header />);
    const text = <h2>Enkh-Erdene Bolormaa</h2>;

    expect(header.contains(text)).toEqual(true);
  });

  it('renders h3', () => {
    const header = shallow(<Header />);
    const text = <h3>Full-Stack Web Developer</h3>;

    expect(header.contains(text)).toEqual(true);
  });
});
