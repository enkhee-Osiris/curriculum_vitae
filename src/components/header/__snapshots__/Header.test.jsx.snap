// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          src="image.png"
          width="200px"
/>,
        <h2>
          Enkh-Erdene Bolormaa
</h2>,
        <h3>
          Full-Stack Web Developer
</h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "src": "image.png",
          "width": "200px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enkh-Erdene Bolormaa",
        },
        "ref": null,
        "rendered": "Enkh-Erdene Bolormaa",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Full-Stack Web Developer",
        },
        "ref": null,
        "rendered": "Full-Stack Web Developer",
        "type": "h3",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            src="image.png"
            width="200px"
/>,
          <h2>
            Enkh-Erdene Bolormaa
</h2>,
          <h3>
            Full-Stack Web Developer
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "src": "image.png",
            "width": "200px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enkh-Erdene Bolormaa",
          },
          "ref": null,
          "rendered": "Enkh-Erdene Bolormaa",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Full-Stack Web Developer",
          },
          "ref": null,
          "rendered": "Full-Stack Web Developer",
          "type": "h3",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
